src/box/sql/func.c

```diff
@@ -2414,40 +2414,33 @@ func_sql_expr_call(struct func *func, struct port *args, struct port *ret)
 {
 	struct func_sql_expr *func_sql = (struct func_sql_expr *)func;
 	struct sql_stmt *stmt = (struct sql_stmt *)func_sql->stmt;
-	const char *data;
-	uint32_t mp_size;
-	struct tuple_format *format;
-	if (args->vtab == &port_c_vtab && ((struct port_c *)args)->size == 2 &&
-	    ((struct port_c *)args)->first_entry.mp_format != NULL) {
-		/*
-		 * The only case where mp_format is not NULL is when the
-		 * function is used as a CHECK constraint.
-		 */
-		struct port_c_entry *pe = ((struct port_c *)args)->first;
-		data = pe->mp;
-		mp_size = pe->mp_size;
-		format = pe->mp_format;
-	} else {
+	if (args->vtab != &port_c_vtab || ((struct port_c *)args)->size != 2) {
 		diag_set(ClientError, ER_UNSUPPORTED, "Tarantool",
 			 "SQL functions");
 		return -1;
 	}
-
 	struct region *region = &fiber()->gc;
 	size_t svp = region_used(region);
 	port_sql_create(ret, stmt, DQL_EXECUTE, false);
+	struct port_c_entry *pe = ((struct port_c *)args)->first;
 	/*
-	 * In SQL, we can only retrieve fields that have names. There is no
-	 * point to prepare slots for nameless fields.
+	 * Currently, SQL EXPR functions can only be called in a tuple or field
+	 * constraint. If the format is NULL then it is a field constraint,
+	 * otherwise it is a tuple constraint.
 	 */
-	uint32_t count = format->min_field_count;
-	struct vdbe_field_ref *ref;
-	size_t size = sizeof(ref->slots[0]) * count + sizeof(*ref);
-	ref = region_aligned_alloc(region, size, alignof(*ref));
-	vdbe_field_ref_prepare_data(ref, data, mp_size);
-	ref->format = format;
-	if (sql_bind_ptr(stmt, 1, ref) != 0)
-		goto error;
+	if (pe->mp_format != NULL) {
+		uint32_t count = pe->mp_format->total_field_count;
```

Comment:
I see in the old code it was `min_field_count`. I remember this was fixed separately somewhere, no?

Reply:
You mentioned this previously: https://github.com/tarantool/tarantool/pull/7704#discussion_r981782286

Is that what you're asking?

Reply:
Yeah. This looks like a separate bug, no? I can't see how `min_field_count` -> `total_field_count` change is related to introduction of `SQL_EXPR` functions. Am I missing something?

Reply:
You are right. I moved this change to a new patch.